### ISSUE TITLE
Paravalidators authoring blocks but missing all votes

### ISSUE THREAD

**Issue by SimonKraus** [paritytech/polkadot-sdk#4991]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

This issue is going on for some months already now on both networks - Polkadot and Kusama. 

It has been discussed some times in Matrix and @bLd75 dropped something related to [stackexchange](https://substrate.stackexchange.com/questions/11397/paravalidator-session-keys-requires-read-permission-for-non-owner).

Problem:
Validators are proposing relaychain blocks that are accepted.
The Validator in does not vote on any parachain blocks (neither implicit nor explicit) and misses all votes resulting in 0 backing points.

At the time of this happening there are no weird metrics going on on a quick glance and there's nothing suspicious in the logs.

![KusamaValidatorsNotParavoting](https://github.com/paritytech/polkadot-sdk/assets/5071303/41df37f0-3a30-43aa-9595-471837d9a05b)

Currently this affects 20 Validators in the active Kusama sets according to [Turboflakes Dashboard](https://apps.turboflakes.io/?chain=kusama#/insights).

I've myself experienced this issue multiple times and spoke to many fellow operators and despite some things work for in some cases (like restarting the node, rotating the keys, changing key permissions) this issue is still very much unpredictable to me.

### Steps to reproduce

unverifiableRandomnessFunction()

**Comment by alexggh** [comment]
I've been tracking this type of bugs down, some failure modes that we have been observing in the order of impact:

1. Validators running pre-async backing polkadot versions. [Nothing can be done here, they will get 0 points]
2. Validators are not reachable  by the rest of validators/collators, because they change public IP, PeerId. [Nothing can be done here, they recover by themselves after the new records propagate through the system]

I'm not excluding the possibility of some other eluding bug, can you provide us with a validator that you control where this happened  and all the logs around that time, this should help us understand if it is one of the above or maybe something else.

**Comment by SimonKraus** [comment]
> 2. Validators are not reachable  by the rest of validators/collators, because they change public IP, PeerId. [Nothing can be done here, they recover by themselves after the new records propagate through the system]

I've seen this "automatically resolving" on some nodes without manual interaction, is there an automatic peerId-reneval somehow without renewing the network secret or restarting the node? 

> I'm not excluding the possibility of some other eluding bug, can you provide us with a validator that you control where this happened and all the logs you can provide around that time(the more the merrier), this should help us understand if it is one of the above or maybe something else.

I'll ask Saxemberg who's currently having these kinds of issues 🙏 

Thanks for looking into it

**Comment by SimonKraus** [comment]
I can tell that in my case it was not restarted and IP change can't happen (own /26 subnet in my colocation rack).

The startup command was pretty basic:
```
ExecStart=/var/lib/polkadot/polkadot \
     --validator \
     --state-pruning 1000 \
     --blocks-pruning 1000 \
     --database paritydb \
     --base-path /var/lib/polkadot \
     --chain polkadot \
     --prometheus-external 
```

And it happened on latest release.

**Comment by m-saxemberg** [comment]
We want to report the same issue on one of our Kusama validators. 
The command was similar to Simon's.
We tried some proposed solutions like setting new permissions as well as just waiting for the automatic resolution but they didn't work. 
We have the chain file available upon request if that's required. We are currently in touch with the data center where our Kusama nodes are colocated because we have a suspicion of a ddos attack on their infrastructure which might have something to do with our issues but we don't have confirmation of that just yet.

**Comment by alexggh** [comment]
@m-saxemberg
To help us understand what is going on can you share:
1. Link to your validator in app.turboflakes.
2. All logs you have for the said validator.
3. What polkadot version you are running.
4. Anything else you consider relevant regarding operating the validator(restarts, setup, etc.)

**Comment by paradox-tt** [comment]
Might as well note that I'm seeing the same on my Kusama validators, most recent as yesterday.  In my case a restart helps.

**Comment by alexggh** [comment]
@paradox-tt  can you share the information from the above, the more logs you have the better.

**Comment by CertHum-Jim** [comment]
Seeing this right now in our Kusama validator [Turboflakes](https://apps.turboflakes.io/?chain=kusama#/validator/FVwmuHZRbnEBrA6ya4SYAGt9ChtZWqqjgF2H758mAcHMhuW)

This validator was working fine. Yesterday I rebooted the server (Ubuntu 22.04 LTS), and the service failed to start. I needed to regenerate the network key and used ```--unsafe-force-node-key-generation \``` option. I restarted again after that with the flag removed and it started fine. I'll try to get some logs and leave it misbehaving for now so anyone can see.

**Comment by CertHum-Jim** [comment]
[Turboflakes](https://apps.turboflakes.io/?chain=kusama#/validator/FVwmuHZRbnEBrA6ya4SYAGt9ChtZWqqjgF2H758mAcHMhuW)

@alexggh this node is still not paravalidating after 72hrs+. Is there a recommended course of action?

**Comment by mchaffee** [comment]
This has dinged us (LuckyFriday) several times – most recently today. Prepping logs to post.

**Comment by mchaffee** [comment]
[log.txt](https://github.com/user-attachments/files/16483264/log.txt)

**Comment by SimonKraus** [comment]
additionally I don't think that https://github.com/paritytech/polkadot-sdk/commit/6720279fb3aac12ad525785d2366cdf30da4d78c will solve the issue ultimately, as most of the people affected by this issue DID NOT delete/regenerate their node secret or change their IP.

It seems most likely to happen if a node re-enters from the waiting list (so it's frequently affecting 1KV validators)

**Comment by alexggh** [comment]
> [log.txt](https://github.com/user-attachments/files/16483264/log.txt)

@mchaffee your **polkadot** validator https://apps.turboflakes.io/?hain=polkadot#/validator/14AakQ4jAmr2ytcrhfmaiHMpj5F9cR6wK1jRrdfC3N1oTbUz, seems to be getting load of points now did you do anything to fix it I assume just the restarts I see in the logs ?.

```
authority="14DsoQ7hnQFPs2waL7MMRFhQJanWtkzkygKuYcrbWcvmy2Ld" peer_id=PeerId("12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2") addresses="Parity Polkadot/v1.15.0-743dc632fd6 (🍀LuckyFriday-EU-DOT-01🍀)" version={"/ip4/10.48.113.226/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.12.148/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/100.64.0.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.148.14.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.50.110.6/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.10.14/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.97.240/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.118.255/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.16.0.23/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/127.0.0.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.109.61/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.19.0.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.3.85/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.19.10.59/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.125.137/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.101.247/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/100.64.4.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.126.8/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.23.128/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.102.139/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.103.74/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/199.247.24.97/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.108.27/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.16.51.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.127.148/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.124.121/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.102.155/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.19.13.15/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.15.136/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.23.0.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.116.95/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.98.169/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.1.88/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.96.220/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.104.66/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.12.202/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.117.135/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.127.147/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.7.62/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.29.0.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.97.240/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.116.128/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/100.64.2.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.125.111/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.111.171/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.22.0.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.19.27/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/100.64.31.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.101.59/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.16.16.10/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.148.13.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.6.232/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.138.125.59/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.16.0.43/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.116.6/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/100.64.1.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/192.168.50.50/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.118.93/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.122.35/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.19.238/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.50.110.5/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.119.181/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.113.53/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.117.239/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.113.148/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.119.90/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.106.253/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.47.10.62/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/172.19.13.51/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.48.116.39/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.148.12.1/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2", "/ip4/10.129.7.110/tcp/30330/p2p/12D3KooWSEQZzFjj7R57NMgLFjphPPT9KVqrd8kiyvf6XBckWUH2"}  num_exceeded 72 raw 8e86a73630766eeb5df898e509269083fca8dafcd6c821bec17717fa0ff97a24
```

Anyways I think your problem is caused by the fact that your validator seem to be reporting 73 different IP address, that is bound to create problems with connectivity between validators, not sure how you got into this state, probably the information is in the logs before you join the active set, I would expect you would see a lot of:
```
Discovered new external address for our node
```

FYI @paritytech/networking, maybe you got an idea how the node might got into this state, I've think we seen this happening on rococo here: https://github.com/paritytech/polkadot-sdk/issues/3519#issuecomment-1994760856.

Do you by any chance run more than on node which might connected between each other using those addresses ?

I think after the restart your node cleared all those accumulated IP address and that's why it is working now.

**Comment by alexggh** [comment]
> [Turboflakes](https://apps.turboflakes.io/?chain=kusama#/validator/FVwmuHZRbnEBrA6ya4SYAGt9ChtZWqqjgF2H758mAcHMhuW)
> 
> @alexggh this node is still not paravalidating after 72hrs+. Is there a recommended course of action?

@CertHum-Jim looked at your `Kusama` validator and after running https://github.com/lexnv/subp2p-explorer, it seems like I can't reach it:
```
authority="GCuezmyWaSubhgjWssjvLqo7sXcGJYadW5DfNTsdje9qtrz" peer_id=PeerId("12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa") addresses={"/ip4/15.235.216.1/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa", "/ip4/103.240.197.70/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa", "/ip6/2402:1f00:8001:2201::/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa"} "num 3" raw a09c7441c8db86dafd48e04be28fc6691ecd8d6bf1b32e85db93937b11627125 - Cannot be reached
```

Given your validator gets 0 *backing points* and 0 *points* for authoring relay chain blocks, I would check if you validator is connected and in sync with the relay chain, could you also provide all the logs that you've got and I can try to advice based on that.

**Comment by alexggh** [comment]
> additionally I don't think that [6720279](https://github.com/paritytech/polkadot-sdk/commit/6720279fb3aac12ad525785d2366cdf30da4d78c) will solve the issue ultimately, as most of the people affected by this issue DID NOT delete/regenerate their node secret or change their IP.
> 
> It seems most likely to happen if a node re-enters from the waiting list (so it's frequently affecting 1KV validators)

Yes, the mentioned commit won't help if addresses did not change or if they are invalid or if there is some other bug at play, because that helps just with propagating newer addresses faster.

> It seems most likely to happen if a node re-enters from the waiting list (so it's frequently affecting 1KV validators)

Could you please direct them on this ticket with their logs.

Unfortunately this issue is generic enough and happens occasionally. Anything wrong with the validators will manifest as missing points, so unless we've got logs and specific details about the setup and network condition at the time it happens we won't be able to properly root-cause it.

**Comment by CertHum-Jim** [comment]
> > [Turboflakes](https://apps.turboflakes.io/?chain=kusama#/validator/FVwmuHZRbnEBrA6ya4SYAGt9ChtZWqqjgF2H758mAcHMhuW)
> > @alexggh this node is still not paravalidating after 72hrs+. Is there a recommended course of action?
> 
> @CertHum-Jim looked at your `Kusama` validator and after running https://github.com/lexnv/subp2p-explorer, it seems like I can't reach it:
> 
> ```
> authority="GCuezmyWaSubhgjWssjvLqo7sXcGJYadW5DfNTsdje9qtrz" peer_id=PeerId("12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa") addresses={"/ip4/15.235.216.1/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa", "/ip4/103.240.197.70/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa", "/ip6/2402:1f00:8001:2201::/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa"} "num 3" raw a09c7441c8db86dafd48e04be28fc6691ecd8d6bf1b32e85db93937b11627125 - Cannot be reached
> ```
> 
> Given your validator gets 0 _backing points_ and 0 _points_ for authoring relay chain blocks, I would check if you validator is connected and in sync with the relay chain, could you also provide all the logs that you've got and I can try to advice based on that.

The node is online, the chain is synced, but the peer ID that is seen in p2p is not what is on the node (and you can see the correct one in telemetry and part of 1KV -- and when it change in the pastbin logs, when i originally reported the bug.)

Also, in your output, the first IPv4 address and the third are correct. The second I have never seen before and is not from any provider we've ever used -- it's really strange that it is seeing that on the network.

https://pastebin.com/t3FDFigr

**Comment by CertHum-Jim** [comment]
> > [Turboflakes](https://apps.turboflakes.io/?chain=kusama#/validator/FVwmuHZRbnEBrA6ya4SYAGt9ChtZWqqjgF2H758mAcHMhuW)
> > @alexggh this node is still not paravalidating after 72hrs+. Is there a recommended course of action?
> 
> @CertHum-Jim looked at your `Kusama` validator and after running https://github.com/lexnv/subp2p-explorer, it seems like I can't reach it:
> 
> ```
> authority="GCuezmyWaSubhgjWssjvLqo7sXcGJYadW5DfNTsdje9qtrz" peer_id=PeerId("12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa") addresses={"/ip4/15.235.216.1/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa", "/ip4/103.240.197.70/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa", "/ip6/2402:1f00:8001:2201::/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa"} "num 3" raw a09c7441c8db86dafd48e04be28fc6691ecd8d6bf1b32e85db93937b11627125 - Cannot be reached
> ```
> 
> Given your validator gets 0 _backing points_ and 0 _points_ for authoring relay chain blocks, I would check if you validator is connected and in sync with the relay chain, could you also provide all the logs that you've got and I can try to advice based on that.

Just a little more info, the 103.240.197.70 IP comes from an AS run by Leapswitch and the block seems like it is assigned to Sunday Networks, a HK company. Traceroutes indicate it is located in India. So I'm not really sure how the same network ID can live in both Singapore and India at the same time.

**Comment by alexggh** [comment]
> The node is online, the chain is synced, but the peer ID that is seen in p2p is not what is on the node (and you can see the correct one in telemetry and part of 1KV -- and when it change in the pastbin logs, when i originally reported the bug.)

What do you mean is not correct in both dht and your logs I see `12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa`  which seems correct to me.

 >Also, in your output, the first IPv4 address and the third are correct. The second I have never seen before and is not from any provider we've ever used -- it's really strange that it is seeing that on the network.
 
 You are saying this one: `"/ip4/103.240.197.70/tcp/11002/p2p/12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa"` is not correct although I see it in your logs, maybe it is some address your node automatically resolves to ? 
 ```
  Discovered new external address for our node: /ip4/103.240.197.70/tcp/11002/p2p/12D3KooWFsxi9vj5HVMm5iFLS6b5C9JSagmWfFRnAGKSsSjJRTGd
 ```
 
 @CertHum-Jim 
 
1. Do you have more logs than that, like everything since the issue started till now(the more the better) ?
2. Can you start your node with this logs enabled, just append `parachain::gossip-support=debug,sub-authority-discovery=debug`  to this line in your command line `-lsync=warn,afg=warn,babe=warn`

**Comment by CertHum-Jim** [comment]
> @CertHum-Jim
> 
> 1. Do you have more logs than that, like everything since the issue started till now(the more the better) ?
> 2. Can you start your node with this logs enabled, just append `parachain::gossip-support=debug,sub-authority-discovery=debug`  to this line in your command line `-lsync=warn,afg=warn,babe=warn`

File is here: https://drive.google.com/file/d/1NN3lZyaPdiFu7TIeExjW_VuFeE_eiAL-/view?usp=sharing

Maybe I'm just reading it wrong, but that 103. address sure does seems like it's associated with a lot of peers.

**Comment by alexggh** [comment]
> Maybe I'm just reading it wrong, but that 103. address sure does seems like it's associated with a lot of peers.

Yes, indeed, but I think that's a read herring, because all other nodes work, I think the main problem is why all authorithies still see your old PeerId `12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa` and not the new one `12D3KooWFsxi9vj5HVMm5iFLS6b5C9JSagmWfFRnAGKSsSjJRTGd`, the old records should have expired by now. 

I'll try to figure it out what is happening tomorrow, till then I guess your best try to recover this validator is to change your authority key by doing a key rotation, which I found to be helpful in this kind of cases.

**Comment by CertHum-Jim** [comment]
> > Maybe I'm just reading it wrong, but that 103. address sure does seems like it's associated with a lot of peers.
> 
> Yes, indeed, but I think that's a read herring, because all other nodes work, I think the main problem is why all authorithies still see your old PeerId `12D3KooWLmo8ohNty4QhFMAyF6a5CKf4jFL5A9kjC56iaWHfCrQa` and not the new one `12D3KooWFsxi9vj5HVMm5iFLS6b5C9JSagmWfFRnAGKSsSjJRTGd`, the old records should have expired by now.
> 
> I'll try to figure it out what is happening tomorrow, till then I guess your best try to recover this validator is to change your authority key by doing a key rotation, which was found to be helpful in this kind of cases.

Ok, thanks for looking into it. Also, i hijacked this issue a bit because I think the original report from @SimonKraus occurred without the network ID being changed on the node, so just adding that to put it back on topic (although maybe the root is related).

We did have that happen once  (where no changes were made, just out and in of the set) on our TVP polkadot node and we found the only remedy to deploy a new node with a new network and authority key. But for this instance I'll just rotate and hopefully will start working.

**Comment by validorange** [comment]
I found this issue on a google search... My Polkadot validator (https://apps.turboflakes.io/?chain=polkadot&app=onet#/validator/12WRpiUFDZ1GjksnpmBL3EzT2jz2TZ7QHhUTqoubhYPQ7PAZ?mode=history) has been getting block points but no backing points. This started when the node became active in session 11046, despite working without issue the previous time it was active which was session 11015.

I enabled debug logging (although I lost some logs to rate limiting) so I am happy to send whatever might be useful.

The validator became active again in session 11070 and is still having the same issue, despite restarting the service. I have just now rotated keys so I will see if that helps.  I am also working on preparing another server so I can cut over to a new server if rotating keys does not fix it.

**Comment by validorange** [comment]
Here is a log bundle: https://drive.google.com/file/d/1pJh0kA33QBryG17QvFFl2HHf1AGyn0bY/view?usp=sharing

I am not really sure if adding to this old issue is best or if I should open a new one. If this should be a new issue let me know and I will open one.

**Comment by drskalman** [comment]
We moved some of our validators to servers with better specs and it is more than two days that they are (neither Polkadot Nor Kusama) are not making any backing points only block production. how many days worth of reward we are supposed to lose before the parachains find out about our new ips? Should we move back to old machines till the network find out about the new machines in a week or month or? If so, shouldn't there be a warning in validator setup guide about the serious financial implications of changing machine and the recommendation of not changing to a machine which has not been part of the network for an extended period of time (like a week, a month, etc I don't know what is the expected prorogation time, but clearly 3 days has not been enough).

**Comment by alexggh** [comment]
> We moved some of our validators to servers with better specs and it is more than two days that they are (neither Polkadot Nor Kusama) are not making any backing points only block production. how many days worth of reward we are supposed to lose before the parachains find out about our new ips? Should we move back to old machines till the network find out about the new machines in a week or month or? If so, shouldn't there be a warning in validator setup guide about the serious financial implications of changing machine and the recommendation of not changing to a machine which has not been part of the network for an extended period of time (like a week, a month, etc I don't know what is the expected prorogation time, but clearly 3 days has not been enough).

Definitely not 3 days, it should propagate to everyone in a few hours, so if it did not work for 3 days, I suspect a different issue.

**Comment by Paride1** [comment]
Good morning,
I have the same issue, no points for parachain but only for blocks.
My stash is: 143bj6p3rQeoC1RuWEgZNu2pbwaQtdhiwX28599EdqvCSPmb 
I have already rotated the keys (2 sessions pass), restarted the process and restarted the server.
The version is the last 1.19.2, the server is located in Mumbai.
Any ideas?
Thanks

**Comment by alexggh** [comment]
> Good morning, I have the same issue, no points for parachain but only for blocks. My stash is: 143bj6p3rQeoC1RuWEgZNu2pbwaQtdhiwX28599EdqvCSPmb I have already rotated the keys (2 sessions pass), restarted the process and restarted the server. The version is the last 1.19.2, the server is located in Mumbai. Any ideas? Thanks

It seems like some validators in the network can't discover your validator:
```
 Couldn't resolve addresses of authority: Public(424ae16c3226f2cd9527c3be296cbf4bc51ad5f79733020d0507506f5f68ea1a (12VvQTwx...))
```
I expect you are hitting a flavour of this issue: https://github.com/paritytech/polkadot-sdk/issues/8548, can you share logs and what actions you might have taken.

**Comment by Paride1** [comment]
> > Good morning, I have the same issue, no points for parachain but only for blocks. My stash is: 143bj6p3rQeoC1RuWEgZNu2pbwaQtdhiwX28599EdqvCSPmb I have already rotated the keys (2 sessions pass), restarted the process and restarted the server. The version is the last 1.19.2, the server is located in Mumbai. Any ideas? Thanks
> 
> It seems like some validators in the network can't discover your validator:
> 
> ```
>  Couldn't resolve addresses of authority: Public(424ae16c3226f2cd9527c3be296cbf4bc51ad5f79733020d0507506f5f68ea1a (12VvQTwx...))
> ```
> 
> I expect you are hitting a flavour of this issue: [#8548](https://github.com/paritytech/polkadot-sdk/issues/8548), can you share logs and what actions you might have taken.

My node was active in DN 2 in another location and it left the set of active validator during DN 2.1. During DN 2.1 I migrated to Mumbai for DN 3 and today is the first day of DN 3. 
I created a new key for first power up but yesterday I found out that the node had 0 points after one session.
I checked the key and author_hasSessionKey return false, so I created a new key and did a rotate. After 2 sessions the validator validates blocks but not parachain.
After that I restarted process and I restarted server but nothing changes.
Now I tried to use --network-backend libp2p but don't seem to solve.
P2P Port is open and accessible.
I will send the logs in the evening

**Comment by alexggh** [comment]
> Now I tried to use --network-backend libp2p but don't seem to solve.

I would run it with the default litep2p since that seems to be less affected by https://github.com/paritytech/polkadot-sdk/issues/8548.

**Comment by Paride1** [comment]
I checked #8548, 

> > Now I tried to use --network-backend libp2p but don't seem to solve.
> 
> I would run it with the default litep2p since that seems to be less affected by [#8548](https://github.com/paritytech/polkadot-sdk/issues/8548).

Ok I'll go back.
In the issues #8548 I read

_Restarting the Polkadot bootnodes, followed by restarting the affected validator — resolved the issue_

How can I do this?

**Comment by alexggh** [comment]
> Restarting the Polkadot bootnodes, 

You can't because I guess you don't own polkadot bootnodes. I see some of our validator found your node, so maybe it is just slow to be discovered because of the location.
```
 Found addresses for authority Public(424ae16c3226f2cd9527c3be296cbf4bc51ad5f79733020d0507506f5f68ea1a (12VvQTwx...)): {"/ip4/148.113.0.140/tcp/30333/p2p/12D3KooWRphBtdHbwFPhyU8jzifdoqdMM5GKs9rZb1PEUDxThe6v", "/ip6/2402:1f00:8300:8c::/tcp/30333/p2p/12D3KooWRphBtdHbwFPhyU8jzifdoqdMM5GKs9rZb1PEUDxThe6v"}
```

Anyways, logs should helps a bit more.

**Comment by Paride1** [comment]
> > Restarting the Polkadot bootnodes,
> 
> You can't because I guess you don't own polkadot bootnodes. I see some of our validator found your node, so maybe it is just slow to be discovered because of the location.
> 
> ```
>  Found addresses for authority Public(424ae16c3226f2cd9527c3be296cbf4bc51ad5f79733020d0507506f5f68ea1a (12VvQTwx...)): {"/ip4/148.113.0.140/tcp/30333/p2p/12D3KooWRphBtdHbwFPhyU8jzifdoqdMM5GKs9rZb1PEUDxThe6v", "/ip6/2402:1f00:8300:8c::/tcp/30333/p2p/12D3KooWRphBtdHbwFPhyU8jzifdoqdMM5GKs9rZb1PEUDxThe6v"}
> ```
> 
> Anyways, logs should helps a bit more.

I'm saving them, are the last two days enough?

**Comment by alexggh** [comment]
> I'm saving them, are the last two days enough?

Yes, ideally you also have some logs where you enable the authority discovery logs, you can do it with, passing `-lsub-authority-discovery=debug` to the command line.

**Comment by Paride1** [comment]
[log_from_mumbai.log.gz](https://github.com/user-attachments/files/22278271/log_from_mumbai.log.gz)

I add _-lsub-authority-discovery=debug_ and I'll send you more logs in a few hours after the restart? OK?

**Comment by alexggh** [comment]
> [log_from_mumbai.log.gz](https://github.com/user-attachments/files/22278271/log_from_mumbai.log.gz)
> 
> I add _-lsub-authority-discovery=debug_ and I'll send you more logs in a few hours after the restart? OK?

Sep 11 03:39:52 ns5023820 polkadot[554060]: 2025-09-11 03:39:52 🥩 Multiple private keys found for: [Public(025e2b640c740c2104058a334b8d9b21b7bb6396e7111abbb42d097fa1992f3787 (1EhnufAH...)), Public(02b2f210f14a71cf10cfc35c9be21a12e526e7a5104af0702b844bdb6c4d195607 (1Gcwx3r9...))] (2)

I think you kind of moved the keys of both your validators in the same place, that is bound to produce problems.

https://apps.turboflakes.io/?chain=polkadot#/validator/16CdHjb4nxVwF6uwmPm6A29pc4ubnLiY7UqasMxt7cT9BcoK
https://apps.turboflakes.io/?chain=polkadot#/validator/143bj6p3rQeoC1RuWEgZNu2pbwaQtdhiwX28599EdqvCSPmb?mode=history

**Comment by Paride1** [comment]
Ok, it all started with:

- the node NEWDEAL1 doesn't score points
- the node NEWDEAL works well

I retrieved the onchain keys for the two accounts and saw that for both `author_hasSessionKeys` returns true on NEWDEAL1 and false on NEWDEAL

So I made a purge key for both nodes, created a new key for each and made a rotate key for each node

Now the `author_hasSessionKeys` command correctly returns true only once in the respective node. I hope that after two sessions everything will be fine.
Did I follow the correct procedure?

**Comment by Paride1** [comment]
I tried to remember what I did in the past few months and I remembered that:

1. During DN 2.1 only NEWDEAL was in the validation set and I setup a new node in Mumbai waiting for DN3
2. I generated a new key in Mumbai and I moved NEWDEAL there
3. I changed OS of old server (ex NEWDEAL) and I setup a new node
4. I generated a new key and I came back with NEWDEAL in the old server with new OS
5. Before DN 3 I generated a new key in Munbai and I rotated on NEWDEAL1 account

Now I checked on subscan and I realized that the points above are what I thought I had done but in reality at point 5 I did not generate a new key in Mumbai but I only set the validation for NEWDEAL1 with the old NEWDEAL key

The only thing i don't understand at this point is why when i generated a new key yesterday things didn't work out.

In any case, I hope the purge key procedure I wrote above can solve the problem.

**Comment by Paride1** [comment]
I can confirm that now, after the purge and the new keys, everything works as it should.
Thanks for your help.
Final question: Is the reconstruction I made above plausible?
